src/probes/isRequire.js

```diff
@@ -11,28 +11,50 @@ import {
   getCallExpressionArguments
 } from "@nodesecure/estree-ast-utils";
 
-function validateNode(node, { tracer }) {
+function validateNode(node, option) {
   const id = getCallExpressionIdentifier(node);
+  const argument = getCallExpressionArguments(node);
+
   if (id === null) {
     return [false];
   }
 
-  const data = tracer.getDataFromIdentifier(id);
+  const data = option.tracer.getDataFromIdentifier(id);
+
+  if (data !== null && data.name === "require") {
+    return [
+      true,
+      data?.identifierOrMemberExpr ?? void 0
+    ];
+  }
+  else if (id === "eval" && argument[0] === "require") {
+    return [
+      true,
+      option?.identifiersName[0].name ?? void 0
+    ];
+  }
 
   return [
-    data !== null && data.name === "require",
-    data?.identifierOrMemberExpr ?? void 0
+    false,
+    null
   ];
 }
```

Comment:
You don't need to update the original validateNode, just create two distinct functions. And at the bottom provide them to validateNode as Array.

Reply:
thanks it should be good in latest commit